tests/test_herbstluftwm.py

```diff
@@ -21,6 +22,46 @@ def test_reload(hlwm_process, hlwm):
         assert proc.returncode == 0
 
 
+@pytest.mark.parametrize("with_client", [True, False])
+@pytest.mark.parametrize("explicit_arg", [True, False])
+def test_wmexec_to_self(hlwm, hlwm_process, with_client, explicit_arg):
+    if with_client:
+        winid, _ = hlwm.create_client()
+    # modify some attribute such that we can verify that hlwm re-booted
+    hlwm.attr.settings.snap_gap = 13
+
+    # Restart hlwm:
+    args = [hlwm_process.bin_path, '--verbose'] if explicit_arg else []
+    p = hlwm.unchecked_call(['wmexec'] + args,
+                            read_hlwm_output=False)
+    assert p.returncode == 0
+    hlwm_process.read_and_echo_output(until_stdout='hlwm started')
+
+    assert hlwm.attr.settings.snap_gap() != 13
+    if with_client:
+        assert winid in hlwm.list_children('clients')
+
+
+@pytest.mark.parametrize("with_client", [True, False])
```

Comment:
Do we actually need this parameter? We have to create a client anyway as you said in line 50

Reply:
I've removed the comment about the necessity for a client (and the client itself), because the client isn't necessary here (the client was only necessary in the already existing wmexec test case `test_ewmh::test_tags_restored_after_wmexec` where the preservation of the tag names is tested: there, xvfb clears all X properties if there is no client)

Reply:
Cool, thanks for removingit!